build.ps1

```diff
@@ -0,0 +1,119 @@
+[CmdletBinding(DefaultParameterSetName='RegularBuild')]
+param (
+    [ValidateSet("debug", "release")]
+    [string]$Configuration = 'debug',
+    [int]$BuildNumber,
+    [switch]$SkipRestore,
+    [switch]$SkipSubModules,
+    [switch]$CleanCache,
+    [string]$SimpleVersion = '1.0.0',
+    [string]$SemanticVersion = '1.0.0-zlocal',
+    [string]$PackageSuffix,
+    [string]$Branch,
+    [string]$CommitSHA,
+    [string]$BuildBranch = '81aeee183e8a171e4c0fd29fc78de509339808ee'
+)
+
+Set-StrictMode -Version 1.0
+
+# This script should fail the build if any issue occurs.
+trap {
+    Write-Host "BUILD FAILED: $_" -ForegroundColor Red
+    Write-Host "ERROR DETAILS:" -ForegroundColor Red
+    Write-Host $_.Exception -ForegroundColor Red
+    Write-Host ("`r`n" * 3)
+    exit 1
+}
+
+if (-not (Test-Path "$PSScriptRoot/build")) {
+    New-Item -Path "$PSScriptRoot/build" -ItemType "directory"
+}
+
+# Enable TLS 1.2 since GitHub requires it.
+[Net.ServicePointManager]::SecurityProtocol = [Net.ServicePointManager]::SecurityProtocol -bor [Net.SecurityProtocolType]::Tls12
+
+Invoke-WebRequest -UseBasicParsing -Uri "https://raw.githubusercontent.com/NuGet/ServerCommon/$BuildBranch/build/init.ps1" -OutFile "$PSScriptRoot/build/init.ps1"
+. "$PSScriptRoot/build/init.ps1" -BuildBranch "$BuildBranch"
+
+Function Clean-Tests {
+    [CmdletBinding()]
+    param()
+    
+    Trace-Log 'Cleaning test results'
+    
+    Remove-Item (Join-Path $PSScriptRoot "Results.*.xml")
+}
+    
+Write-Host ("`r`n" * 3)
+Trace-Log ('=' * 60)
+
+$startTime = [DateTime]::UtcNow
+if (-not $BuildNumber) {
+    $BuildNumber = Get-BuildNumber
+}
+Trace-Log "Build #$BuildNumber started at $startTime"
+
+$BuildErrors = @()
+
+Invoke-BuildStep 'Reset all submodules' { Reset-Submodules } `
+    -skip:($SkipSubModules) `
+    -ev +BuildErrors
+
+Invoke-BuildStep "Update NuGetGallery submodule" {
```

Comment:
Why do we need NuGetGallery for Status?

Reply:
Status page pulls in `NuGetGallery` so it can use the layout files.